nixos/modules/programs/foot/default.nix

```diff
@@ -0,0 +1,79 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.foot;
+
+  settingsFormat = pkgs.formats.iniWithGlobalSection { listsAsDuplicateKeys = true; };
+in
+{
+  options.programs.foot = {
+    enable = lib.mkEnableOption "foot terminal emulator";
+
+    package = lib.mkPackageOption pkgs "foot" { };
+
+    settings = lib.mkOption {
+      inherit (settingsFormat) type;
+      default = {
+        globalSection = { };
+      };
+      description = ''
+        Configuration for foot terminal emulator. Further information can be found in foot's man page foot.ini(5).
+      '';
+    };
+
+    theme = lib.mkOption {
+      type = with lib.types; nullOr str;
+      default = null;
+      description = ''
+        Include theme from https://codeberg.org/dnkl/foot/src/branch/master/themes.
+      '';
+    };
+
+    enableBashIntergration = lib.mkEnableOption "foot bash integration";
+    enableFishIntergration = lib.mkEnableOption "foot fish integration";
+    enableZshIntergration = lib.mkEnableOption "foot zsh integration";
```

Comment:
What about enabling all shell integration by default? Almost all modules in home-manager do so.

Reply:
To be clear, I mean `lib.mkEnableOption "foot bash integration" // { enable = true; }`.

Hmm, I only find two NixOS modules with similar options: `programs.nix-index` and `nixos/modules/config/vte.nix`.  The former defaults to `true` while the latter defaults to `false`.

> Some of them enable it when the shell is enabled.

Could you give me some examples?

If you mean `lib.mkEnableOption "foot fish integration" // { enable = config.programs.fish.enable; }`, then it equals to my suggestion.

> Others have options for that.

I do not follow.  Could you give me some examples?

Reply:
> Could you give me some examples?

Examples: direnv, thefuck, autojump.

> If you mean lib.mkEnableOption "foot fish integration" // { enable = config.programs.fish.enable; }, then it equals to my suggestion.

Yep. :)

Reply:
Ah, yes, they are equal...

Reply:
Unfortunately I can't make the CI happy with this change.

Reply:
> To be clear, I mean `lib.mkEnableOption "foot bash integration" // { default = true; example = false; }`.

Hmm, this presumably works.

Reply:
I do not have a strong opinion on this.  If you prefer defaulting to `false`, feel free to restore to that.  If people want to default to `true`, they can do so in a following PR.